stream/src/main/scala/org/apache/pekko/stream/scaladsl/Sink.scala

```diff
@@ -435,6 +436,24 @@ object Sink {
   def foldAsync[U, T](zero: U)(f: (U, T) => Future[U]): Sink[T, Future[U]] =
     Flow[T].foldAsync(zero)(f).toMat(Sink.head)(Keep.right).named("foldAsyncSink")
 
+  /**
+   * A `Sink` that will invoke the given function for every received element, giving it its previous
+   * output (or the given `false`) and the element as input.
+   * * The returned [[scala.concurrent.Future]] will be completed with value of the final
+   * function evaluation when the input stream ends, or completed with `Failure`
+   * if there is a failure signaled in the stream.
+   */
+  def forall[T](predicate: T => Boolean): Sink[T, Future[Boolean]] = {
+    var acc = true
+    Flow[T].collectWhile {
+      case element if acc && predicate(element) => true
+      case _ if acc                             => acc = false; false
```

Comment:
As the graph can be materialize multi time , this can cause a problem. when you run this graph several times.

Maybe you can try with `statefulMap`.

Reply:
let me a try